packages/forma-36-react-components/src/components/SelectField/SelectField.tsx

```diff
@@ -39,101 +40,94 @@ const defaultProps: Partial<SelectFieldProps> = {
   required: false,
 };
 
-export class SelectField extends Component<SelectFieldProps, SelectFieldState> {
-  static defaultProps = defaultProps;
-
-  state = { value: this.props.value };
-
-  UNSAFE_componentWillReceiveProps(nextProps: SelectFieldProps) {
-    if (this.props.value !== nextProps.value) {
-      this.setState({
-        value: nextProps.value,
-      });
-    }
-  }
+export const SelectField = (props: SelectFieldProps) => {
+  const {
+    validationMessage,
+    className,
+    children,
+    selectProps,
+    testId,
+    formLabelProps,
+    textLinkProps,
+    labelText,
+    helpText,
+    required,
+    onChange,
+    onBlur,
+    value,
+    name,
+    id,
+    ...otherProps
+  } = props;
+  const [valueState, setValueState] = useState<string | undefined>(value);
 
   // Store a copy of the value in state.
   // This is used by this component when the `countCharacters`
   // option is on
-  handleOnChange = (evt: ChangeEvent) => {
-    this.setState({ value: (evt.target as HTMLSelectElement).value });
-    if (this.props.onChange) {
-      this.props.onChange(evt);
+  const handleOnChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setValueState(e.currentTarget.value);
+    if (onChange) {
+      onChange(e);
     }
   };
 
-  render() {
-    const {
-      validationMessage,
-      className,
-      children,
-      selectProps,
-      testId,
-      formLabelProps,
-      textLinkProps,
-      labelText,
-      helpText,
-      required,
-      onChange,
-      onBlur,
-      value,
-      name,
-      id,
-      ...otherProps
-    } = this.props;
+  useEffect(() => {
+    setValueState(value);
+  }, [value]);
```

Comment:
I'm not sure I get what this is necessary for, are there cases where you're not setting the state value to the value it already has?

Reply:
@denkristoffer I had to put this back, TimePicker tests were failing because `value` prop changes doesn't get propagated properly without it. 

it's like `componentWillReceiveProps`, However this is kinda confusing; Shouldn't react functional components re-render automatically on prop value changes? 🤔

Reply:
Yeah I did the same with the Dropdown, but I share your thoughts. It seems like a code smell. Anyway it's fine for now, maybe we can rethink this pattern at some point.